File: Приклад реалізації на інших мовах/task.py
from collections import deque

directions = [(2, 1), (1, 2), (-1, 2), (-2, 1), (-2, -1), (-1, -2), (1, -2), (2, -1)]

# Перевірка, чи можна потрапити в клітину
def is_valid(x, y, board):
    return 0 <= x < 8 and 0 <= y < 8 and not board[x][y]

# Пошук найкоротшого шляху
def find_shortest_path(start, end, blocked):
    visited = [[False] * 8 for _ in range(8)]
    prev = [[None] * 8 for _ in range(8)]
    queue = deque([start])
    visited[start[0]][start[1]] = True

    while queue:
        curr = queue.popleft()

        if curr == end:
            path = []
            while curr != start:
                path.append(curr)
                curr = prev[curr[0]][curr[1]]
            path.append(start)
            path.reverse()
            return len(path) - 1, path

        # Пробуємо всі можливі ходи
        for dir in directions:
            new_x, new_y = curr[0] + dir[0], curr[1] + dir[1]
            if is_valid(new_x, new_y, blocked) and not visited[new_x][new_y]:
                visited[new_x][new_y] = True
                prev[new_x][new_y] = curr
                queue.append((new_x, new_y))

    return -1, []

# Виведення шахової дошки
def print_board(start, end, blocked):
    board = [['.' for _ in range(8)] for _ in range(8)]

    for i in range(8):
        for j in range(8):
            if blocked[i][j]:
                board[i][j] = '#'
    
    board[start[0]][start[1]] = 'S'
    board[end[0]][end[1]] = 'E'

    print("Шахова дошка:")
    for row in board:
        print(" ".join(row))

# Головна функція
def main():
    start_x, start_y = map(int, input("Введіть початкову позицію коня (x1 y1): ").split())
    end_x, end_y = map(int, input("Введіть кінцеву позицію коня (x2 y2): ").split())
    
    blocked_count = int(input("Введіть кількість заблокованих клітин: "))
    blocked = [[False] * 8 for _ in range(8)]
    
    for _ in range(blocked_count):
        bx, by = map(int, input("Введіть заблоковану клітину (x y): ").split())
        blocked[bx-1][by-1] = True

    start = (start_x - 1, start_y - 1)
    end = (end_x - 1, end_y - 1)

    steps, path = find_shortest_path(start, end, blocked)
    if steps == -1:
        print("Неможливо дістатися до кінцевої клітини.")
        return

    history = [f"({p[0]+1}, {p[1]+1})" for p in path]
    print("Історія кроків:", " -> ".join(history))

    print_board(start, end, blocked)

if __name__ == "__main__":
    main()

File: Приклад реалізації на інших мовах/task.cs
using System;
using System.Collections.Generic;

class Program
{
    // Структура для зберігання позицій
    struct Position
    {
        public int x, y;
        public Position(int x, int y)
        {
            this.x = x;
            this.y = y;
        }
    }

    // Напрями для ходів коня
    static Position[] directions = new Position[]
    {
        new Position(2, 1), new Position(1, 2), new Position(-1, 2), new Position(-2, 1),
        new Position(-2, -1), new Position(-1, -2), new Position(1, -2), new Position(2, -1)
    };

    // Перевірка, чи можна потрапити в клітину
    static bool IsValid(int x, int y, bool[,] board)
    {
        return x >= 0 && x < 8 && y >= 0 && y < 8 && !board[x, y];
    }

    // Пошук найкоротшого шляху
    static (int, List<Position>) FindShortestPath(Position start, Position end, bool[,] blocked)
    {
        bool[,] visited = new bool[8, 8];
        Position[,] prev = new Position[8, 8];
        Queue<Position> queue = new Queue<Position>();

        visited[start.x, start.y] = true;
        queue.Enqueue(start);

        while (queue.Count > 0)
        {
            Position curr = queue.Dequeue();

            if (curr.x == end.x && curr.y == end.y)
            {
                List<Position> path = new List<Position>();
                while (curr.x != start.x || curr.y != start.y)
                {
                    path.Insert(0, curr);
                    curr = prev[curr.x, curr.y];
                }
                path.Insert(0, start);
                return (path.Count - 1, path);
            }

            // Пробуємо всі можливі ходи
            foreach (var dir in directions)
            {
                int newX = curr.x + dir.x;
                int newY = curr.y + dir.y;
                if (IsValid(newX, newY, blocked) && !visited[newX, newY])
                {
                    visited[newX, newY] = true;
                    prev[newX, newY] = curr;
                    queue.Enqueue(new Position(newX, newY));
                }
            }
        }

        return (-1, null);  // Якщо шлях не знайдено
    }

    // Виведення дошки
    static void PrintBoard(Position start, Position end, bool[,] blocked)
    {
        string[,] board = new string[8, 8];

        // Заповнюємо дошку блоками та порожніми клітинами
        for (int i = 0; i < 8; i++)
        {
            for (int j = 0; j < 8; j++)
            {
                if (blocked[i, j])
                    board[i, j] = "#";
                else
                    board[i, j] = ".";
            }
        }

        board[start.x, start.y] = "S";
        board[end.x, end.y] = "E";

        Console.WriteLine("Шахова дошка:");
        for (int i = 0; i < 8; i++)
        {
            for (int j = 0; j < 8; j++)
            {
                Console.Write(board[i, j] + " ");
            }
            Console.WriteLine();
        }
    }

    static void Main(string[] args)
    {
        int startX, startY, endX, endY, blockedCount;

        Console.Write("Введіть початкову позицію коня (x1 y1): ");
        var startPos = Console.ReadLine().Split();
        startX = int.Parse(startPos[0]);
        startY = int.Parse(startPos[1]);

        Console.Write("Введіть кінцеву позицію коня (x2 y2): ");
        var endPos = Console.ReadLine().Split();
        endX = int.Parse(endPos[0]);
        endY = int.Parse(endPos[1]);

        Console.Write("Введіть кількість заблокованих клітин: ");
        blockedCount = int.Parse(Console.ReadLine());

        bool[,] blocked = new bool[8, 8];
        for (int i = 0; i < blockedCount; i++)
        {
            Console.Write("Введіть заблоковану клітину (x y): ");
            var blockedCell = Console.ReadLine().Split();
            int bx = int.Parse(blockedCell[0]);
            int by = int.Parse(blockedCell[1]);
            blocked[bx - 1, by - 1] = true;  // Враховуємо, що координати в C# мають бути від 0
        }

        Position start = new Position(startX - 1, startY - 1);
        Position end = new Position(endX - 1, endY - 1);

        var (steps, path) = FindShortestPath(start, end, blocked);
        if (steps == -1)
        {
            Console.WriteLine("Неможливо дістатися до кінцевої клітини.");
            return;
        }

        List<string> history = new List<string>();
        foreach (var p in path)
        {
            history.Add($"({p.x + 1}, {p.y + 1})");  // Перетворюємо на 1-індексацію
        }

        Console.WriteLine("Історія кроків: " + string.Join(" -> ", history));
        PrintBoard(start, end, blocked);
    }
}
